newIDE/app/src/MainFrame/index.js

```diff
@@ -926,15 +963,25 @@ class MainFrame extends React.Component<Props, State> {
     }: { openEventsEditor: boolean, openSceneEditor: boolean } = {}
   ) => {
     const { i18n, storageProviderOperations } = this.props;
+
+    this.setState({
+      previewFirstSceneName: name,
+    });
+
     const sceneEditorOptions = {
       name,
+      type: 'sceneEditor',
       renderEditor: ({ isActive, editorRef }) => (
         <PreferencesContext.Consumer>
           {({ values }) => (
             <SceneEditor
               project={this.state.currentProject}
               layoutName={name}
               setToolbar={this.setEditorToolbar}
+              isPreviewOverride={this.state.isPreviewOverride}
+              togglePreviewOverride={this.togglePreviewOverride}
+              previewFirstSceneName={this.state.previewFirstSceneName}
+              setScenePreview={this.setScenePreview}
```

Comment:
What if you do `setScenePreview={() => this.setScenePreview(name)}`?

So that the SceneEditor/SceneEditorContainer just have a function to call.
And this.setScenePreview is called with the scene name, which is very convenient because it's exactly what it needs? :)

Reply:
This seem trouble the behavior on the togglePreviewOverride

Reply:
Then it's togglePreviewOverride that must be fixed :) See my other comment.